Lib/test/test_tools/test_pathfix.py

```diff
@@ -0,0 +1,52 @@
+import os
+import subprocess
+import sys
+import unittest
+from test import support
+from test.test_tools import import_tool, scriptsdir
+
+
+class TestPathfixFunctional(unittest.TestCase):
+    script = os.path.join(scriptsdir, 'pathfix.py')
+
+    def setUp(self):
+        self.temp_file = support.TESTFN
+        self.addCleanup(support.unlink, support.TESTFN)
+
+    def pathfix(self, shebang, pathfix_flags):
+        with open(self.temp_file, 'w', encoding='utf8') as f:
+            f.write(f'{shebang}\n' + 'print("Hello world")\n')
+        subprocess.call([sys.executable, self.script, *pathfix_flags, '-n', self.temp_file])
+        with open(self.temp_file, 'r', encoding='utf8') as f:
+            output = f.read()
+            lines = output.split('\n')
+            self.assertEqual(lines[1:], ['print("Hello world")\n']
+            shebang = lines[0][:-1]  # strip the newline
+            return shebang
+
+    def test_pathfix_keeping_flags(self):
```

Comment:
I would prefer:
* put test_pathfix_without_keeping_flags() code into test_pathfix()
* move test_pathfix_keeping_flags() after test_pathfix()
* add a test in test_pathfix_keeping_flags(): use -k but the shebang doesn't contain any flag (the bug that you just fixed)